docs/tutorial/installing_with_flink_and_spark_cluster.md

```diff
@@ -0,0 +1,328 @@
+
+There are many good tutorials on how to install Ubuntu Server on a virtual box, [here is one of them](http://ilearnstack.com/2013/04/13/setting-ubuntu-vm-in-virtualbox/)
+
+### Required Programs
+
+Assuming the minimal install, there are several programs that we will need to install before Zeppelin, Flink, and Spark.
+
+- git
+- openssh-server
+- OpenJDK 7
+- Maven 3.1+
+
+For git, openssh-server, and OpenJDK 7 we will be using the apt package manager. 
+
+##### git
+From the command prompt:
+
+```
+sudo apt-get install git
+```
+
+##### openssh-server
+
+```
+sudo apt-get install open-ssh server
+```
+
+##### OpenJDK 7
+
+```
+sudo apt-get install openjdk-7-jdk openjdk-7-doc openjdk-7-jre-lib
+```
+
+##### Maven 3.1+
+Zeppelin requires maven version 3.x.  The version available in the repositories at the time of writing is 2.x, so maven must be installed manually. 
+
+Purge any existing versions of maven.
+
+```
+sudo apt-get purge maven maven2
+```
+
+Download the maven 3.3.3 binary.
+
+```
+wget "http://www.us.apache.org/dist/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz"
+```
+
+Unarchive the binary and move to the `/usr/local` directory.
+
+```
+tar -zxvf apache-maven-3.3.3-bin.tar.gz
+sudo mv ./apache-maven-3.3.3 /usr/local
+```
+
+Create symbolic links in `/usr/bin`.
+
+```
+sudo ln -s /usr/local/apache-maven-3.3.3/bin/mvn /usr/bin/mvn
+```
+
+### Installing Zeppelin
+This provides a quick overview of Zeppelin installation from source, however the reader is encouraged to review the [Zeppelin Installation Guide](../install/install.html) 
+
+From the command prompt:
+Clone Zeppelin.
+
+```
+git clone https://github.com/apache/incubator-zeppelin.git
+```
+
+Enter the Zeppelin root directory.
+
+```
+cd incubator-zeppelin
+```
+
+Package Zeppelin.
+
+```
+mvn clean package -DskipTests -Pspark-1.5
+```
+
+`-DskipTests` skips build tests- you're not developing (yet), so you don't need to do tests, the clone version *should* build.
+
+`-Pspark-1.5` tells maven to build a Zeppelin with Spark 1.5.  This is important because Zeppelin has its own Spark interpretter and the versions must ~~jive~~ be the same. 
+**Note:** You can build against any version of Spark that has a Zeppelin build profile available. The key is to make sure you check out the matching version of Spark to build. At the time of this writing, Spark 1.5 was the most recent Spark version available.
+
+**Note:** On build failuers. Having installed Zeppelin close to 30 times now, I will tell you that sometimes the build fails for seemingly no reason. 
+As long as you didn't edit any code, it is unlikely the build is failing because of something you did. What does tend to happen, is some dependency that maven is trying to download
+is unreachable.  If you get a build failure, just retry (that is, run the `mvn clean package -DskipTests -Pspark-1.5` again. If it keeps failing, try again tomorrow or after the weekend 
```

Comment:
I think it's more productive to be specific on build failures here.

I.e we could let people know that:
 1. build failure may happen
 2. to understand the reason - they need to look through maven logs (networking issues on access to remote maven repo is one possible reason)
 3. in case that happens and reason is not clear - they can ask help on community mailing lists, file JIRA with logs attached, etc